### DISCUSSION TITLE
Error in plotting NMDS using OTU data

### DISCUSSION THREAD

**Discussion by abkdom** [vegandevs/vegan#486] [Q&A]
Hello all, 
i need assistance on how to resolve the below issue;

I have been trying to plot an nmds  chart using my otu data but when i run this line of the script ;
```r
##vare.mds_Aerobic <- metaMDS(Aerobicsqrt,)

It kept showing me this error message below:

"Warning message:
In metaMDS(Aerobicsqrt, ) :
  stress is (nearly) zero: you may have insufficient data"
```
*************************************************************
Here is my full script and data attached.
##################################### START ###################################
[Aerobic.txt](https://github.com/vegandevs/vegan/files/6087230/Aerobic.txt)
[foresttype_Aerobic.txt](https://github.com/vegandevs/vegan/files/6087231/foresttype_Aerobic.txt)

```r
library(vegan)
Aerobic<-read.table("Aerobic.txt",header=T)


forest_Aerobic<-read.table("foresttype_Aerobic.txt",header=T)

Aerobicsqrt<-sqrt(Aerobic)
gspebc_Aerobic<-vegdist(Aerobicsqrt,"bray")
as.factor(forest_Aerobic$Location)

vare.mds_Aerobic <- metaMDS(Aerobicsqrt,)

ordiplot(vare.mds_Aerobic, type="n")

points(vare.mds_Aerobic, display = "sites", col = c("orange", "darkblue", "purple", "skyblue", "pink", "blue", "red", "green")[as.numeric(as.factor(forest_Aerobic$Location))],
       pch = c(17)[as.numeric(as.factor(forest_Aerobic$Location))], cex = 1.5)
legend("topright",levels(as.factor(forest_Aerobic$Location)), col = c("orange", "darkblue", "purple", "skyblue", "pink", "blue", "red", "green")
       , pch=19, pt.lwd=2,pt.cex=1.0, cex=1.0)
legend("bottomright",levels(as.factor(forest$Position)), pch=c(17,16,2,1), pt.lwd=2,pt.cex=1.2, cex=1.0)

vare.mds$stress
``` 
############################# END #########################################

**Comment by jarioksa** [comment]
It is not an error message, but a warning. A difference between error messages and warnings is that error messages start with word **`Error`** and warnings start with **`Warning`**. In addition, if you get an error, the execution of the function stops and you won't get results. With warnings the execution continues and you get a result (and with default settings you only get a warning after execution and getting the result). Warnings warn: they tell you that you should proceed with caution and inspect your results – there may be something fishy. In this case it tells you that the result is too perfect: stress is nearly zero. This can happen if you do not have sufficient number of distinct observations (low number of observations, duplicated or nearly duplicated observations, too many dimensions for the size of your data etc.). I didn't try, but it seems that you have eleven observations. This is little, but usually sufficient for NMDS (six is the absolute minimum for unique solution). Then it may be that there are some duplicates or almost-duplicates. These appear as overlaid or densely packed groups of points in your plot.

What is wrong with perfect solutions? Perhaps nothing, but they may not be uniquely defined, but you can get a large number of different but equally good solutions. That is the reason we give a warning.

**Comment by jarioksa** [comment]
I had a look at your data: they were really peculiar. Unlike I assumed, the data are not homogeneous, but extremely heterogeneous. However, there are distinct very dissimilar groups, so that NMDS puts all your 11 sampling units in four groups  {A}, {BFGH}, {CED}, {IJK}. The distances within groups are so small compared to the distances between groups that  putting all points within a group in the same spot gives nearly zero stress. You can see this by looking at the `stressplot()`. With your data it would be better to use metric scaling (in **vegan** `wcmdscale` or `dbrda`).

**Comment by abkdom** [comment]
Hi,

Thank you very much for your response. and explanations.
 The 4 groups you mentioned reflect my actual samples. I  will try
the metric scaling you recommended.

Many thanks,


On Fri, Mar 5, 2021 at 12:12 PM Jari Oksanen <notifications@github.com>
wrote:

> I had a look at your data: they were really peculiar. Unlike I assumed,
> the data are not homogeneous, but extremely heterogeneous. However, there
> are distinct very dissimilar groups, so that NMDS puts all your 11 sampling
> units in four groups {A}, {BFGH}, {CED}, {IJK}. The distances within groups
> are so small compared to the distances between groups that putting all
> points within a group in the same spot gives nearly zero stress. You can
> see this by looking at the stressplot(). With your data it would be
> better to use metric scaling (in *vegan* wcmdscale or dbrda).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vegandevs/vegan/issues/399#issuecomment-791380911>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANUR2IUO6LOMGC3QVAVPKKLTCDDCNANCNFSM4YUGLVTA>
> .
>


-- 
*Best regards*

*Abdulkareem Isah Ahmed*
